python/ray/data/_internal/stats.py

```diff
@@ -297,6 +308,15 @@ def clear_stats_actor_metrics(tags: Dict[str, str]):
     _stats_actor.clear_metrics.remote(tags)
 
 
+def get_dataset_id_from_stats_actor() -> str:
+    global _stats_actor
+    _check_cluster_stats_actor()
+    try:
+        return ray.get(_stats_actor.get_dataset_id.remote())
+    except ray.exceptions.RayError:
+        return uuid4().hex
```

Comment:
let's add a comment that `_stats_actor.get_dataset_id` might fail with Ray client, and getting dataset id is best effort, so return `uuid4` here if failing to get dataset id.

Reply:
It's `ray._private.worker._global_node` that doesn't work with Ray Client, on ray client, we won't be able to emit metrics if we can't get `_global_node.cluster_id`. The try catch here doesn't catch that exception